components/search-content/useContentSearch.tsx

```diff
@@ -1,22 +1,53 @@
-import { useLabelerAgent } from '@/shell/ConfigurationContext'
+import { useAppviewAgent, useLabelerAgent } from '@/shell/ConfigurationContext'
 import { useWorkspaceAddItemsMutation } from '@/workspace/hooks'
+import {
+  AppBskyActorDefs,
+  AppBskyActorSearchActors,
+  AppBskyFeedDefs,
+  AppBskyFeedSearchPosts,
+} from '@atproto/api'
 import { useInfiniteQuery } from '@tanstack/react-query'
 
+type ActorResult = AppBskyActorSearchActors.OutputSchema
+type PostResult = AppBskyFeedSearchPosts.OutputSchema
+export type SearchContentSection = 'people' | 'top' | 'latest'
+type QueryData<S extends SearchContentSection> = S extends 'people'
+  ? ActorResult
+  : PostResult
+
 export const useContentSearch = ({
   term,
   section,
 }: {
   term: string
-  section: string
+  section: SearchContentSection
 }) => {
   const labelerAgent = useLabelerAgent()
+  const appviewAgent = useAppviewAgent()
 
   const { mutate: addToWorkspace } = useWorkspaceAddItemsMutation()
 
-  const searchInfo = useInfiniteQuery({
+  const searchInfo = useInfiniteQuery<QueryData<typeof section>>({
     queryKey: ['searchContent', { term, section }],
     enabled: !!term,
     queryFn: async ({ pageParam }) => {
+      if (section === 'people') {
+        // This is actually never hit, we always have appview configured
+        if (!appviewAgent) {
```

Comment:
Can this really be null?!

Reply:
nope. mostly for bypassing the ts check. technically, without appview url configured, it can be null